clients/src/main/java/org/apache/kafka/common/requests/ProduceResponse.java

```diff
@@ -156,7 +182,11 @@ public PartitionResponse(Errors error, long baseOffset, long logAppendTime, long
         }
 
         public PartitionResponse(Errors error, long baseOffset, long logAppendTime, long logStartOffset, List<RecordError> recordErrors, String errorMessage) {
-            this(error, baseOffset, INVALID_OFFSET, logAppendTime, logStartOffset, recordErrors, errorMessage);
+            this(error, baseOffset, INVALID_OFFSET, logAppendTime, logStartOffset, recordErrors, errorMessage, new ProduceResponseData.LeaderIdAndEpoch());
+        }
+
+        public PartitionResponse(Errors error, long baseOffset, long lastOffset, long logAppendTime, long logStartOffset, List<RecordError> recordErrors, String errorMessage) {
+            this(error, baseOffset, lastOffset, logAppendTime, logStartOffset, recordErrors, errorMessage, new ProduceResponseData.LeaderIdAndEpoch());
```

Comment:
can we remove the ProduceResponseData prefixes here?

Reply:
I think I addressed this, the currentLeader parameter was unused so I removed it and set it to be a new LeaderIdAndEpoch by default. I also removed the prefix.